PKHeX/Legality/Checks.cs

```diff
@@ -2503,7 +2503,7 @@ private CheckResult[] parseMovesWasEggPreRelearn(int[] Moves, List<int>[] validL
                 for (int i = 0; i <= splitctr; i++)
                 {
                     var LvlupEggMoves = Legal.getBaseEggMoves(pkm, i, ver, 100)?.ToArray() ?? new int[0];
-                    var EggMoves = !pkm.WasEventEgg && !pkm.WasGiftEgg ? Legal.getEggMoves(pkm, i, ver).ToArray() : new int[0];
+                    var EggMoves = Legal.getEggMoves(pkm, i, ver).ToArray();
```

Comment:
What conflict have you found?
I thinks this code should not be removed, egg moves are not available for event egg and gift egg, because they are generated with a fixed move set that should match the encounter exactly.

If an event egg have a move that is also a egg move from that species then the move should be added to the encounter to allow that egg to have that specific egg move but not all the other egg moves the species could learn